Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/ContestsPage.tsx

```diff
@@ -1,30 +1,54 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import { useSearchParams } from 'react-router-dom';
 import { isNil } from 'lodash';
 import ContestFilters from '../../components/contests/contests-filters/ContestFilters';
 import { useContests } from '../../hooks/use-contests';
 import { setLayout } from '../shared/set-layout';
 import styles from './ContestsPage.module.scss';
+import { IIndexContestsType } from '../../common/types';
+import ContestCard from '../../components/home-contests/contest-card/ContestCard';
+import List from '../../components/guidelines/lists/List';
+import PaginationControls from '../../components/guidelines/pagination/PaginationControls';
 
 const ContestsPage = () => {
-    const { state: { filters } } = useContests();
+    const { 
+        state: {
+            contests,
+            filters
+        },
+        actions: {
+            setPage,
+        },
+        pagesCount,
+    } = useContests();
 
     // TODO: this will be fixed in next PR
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const [ searchParams, setSearchParams ] = useSearchParams();
 
+    const renderContest = useCallback(
+        (contest: IIndexContestsType) => (
+            <ContestCard contest={contest} />
+        ),
+        [],
+    );
+
+    const onPageChange = (event: React.ChangeEvent<unknown>, page: number) => {
```

Comment:
This `event` should be abstracted into `PaginationControls`. Most of the cases, we will not use it